### PR TITLE
refactor mp.

### PR DESCRIPTION
**Pull request by wuhuachaocoding**
### PR types

Others
### PR changes

Others
### Describe

refactor mp:
    this PR mainly changes some dir of mp functions and classes.

- move some private functions about mp to 'fleet/layers/mpu' from collective.py(to simplify it).

-  when using functions for mp operations, we recommend to call it as 'fleet.layers.mpu.xxx'.

users are not aware of it, Compatible with the upgrade.

### CODE REVIEW

File: python/paddle/distributed/fleet/layers/mpu/__init__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from .mp_layers import RowParallelLinear
+from .mp_layers import ParallelCrossEntropy
+
+from .random import RNGStatesTracker
+from .random import get_rng_state_tracker
+from .random import model_parallel_random_seed
+from .random import determinate_seed
+from .random import dropout
+
+from .mp_ops import split
```
Comment (gongweibao) [inline python/paddle/distributed/fleet/layers/mpu/__init__.py]: User need not get these.

**Review by gongweibao — CHANGES_REQUESTED** [review]
Some comments.

File: python/paddle/distributed/fleet/layers/mpu/__init__.py (hunk 1 above)
Comment (wuhuachaocoding) [reply python/paddle/distributed/fleet/layers/mpu/__init__.py]: DONE

File: python/paddle/distributed/fleet/layers/mpu/mp_layers.py [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+from . import mp_ops
+from paddle.fluid import core
+from paddle.fluid.dygraph.layers import Layer
+from .random import get_rng_state_tracker
+from paddle.nn import functional as F
+from paddle import framework
+from ...base import topology as tp
+from paddle.autograd import PyLayer
+
+__all__ = []
```
Comment (gongweibao) [inline python/paddle/distributed/fleet/layers/mpu/mp_layers.py:25]: useful?

**Review by gongweibao — DISMISSED** [review]
LGTM

File: python/paddle/distributed/fleet/layers/mpu/mp_layers.py (hunk 2 above)
Comment (wuhuachaocoding) [reply python/paddle/distributed/fleet/layers/mpu/mp_layers.py:25]: when using functions for mp operations, we recommend to call it as 'fleet.layers.mpu.xxx'.

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**